syntax = "proto3";
package mmo.pb;
option csharp_namespace="Pb";

// sync player`s PID
message SyncPid {
	int32 Pid = 1;
}

// Represent a position
message Position {
	float X = 1;
	float Y = 2;
	float Z = 3;
	float V = 4;
}

// Broadcast message
message BroadCast {
	int32 Pid = 1;
	int32 Tp = 2;
	oneof Data {
        string Content = 3;
        Position P = 4;
		int32 ActionData = 5;
	}
}

// Represent a player
message Player {
	int32 Pid = 1;
	Position P = 2;
}

// Collect plays (info) for sync 
message SyncPlayers {
	repeated Player ps = 1;
}

// Represent a talk action
message Talk {
	string Content=1;
}